### ISSUE TITLE
Public API clarification for stubs

### ISSUE THREAD

**Issue by SeparateRecords** [jiaaro/pydub#566]
Hi there, thanks for this amazing library.

I'm the maintainer of [pydub-stubs](github.com/SeparateRecords/pydub-stubs), and I wanted to get some clarification on what is/isn't part of the public API so I can include it all when updating to v0.25.0.

1. Are all functions from `pydub.utils` intended to be public, or just `register_pydub_effect`?
2. Are `pydub.pyaudioop` and `pydub.logging_utils` public? My guess is no, but I wanted to check just in case :)

**Comment by jiaaro** [comment]
1. I realize this isn't especially helpful, but the way I think about `pydub.utils` is that anyone is free to use those things, and many people do, but the only functions pydub promises to maintain backward compatibility on are ones that are included in the documentation. That's basically just AudioSegment. Having said that, I try never to make breaking changes to anything pydub ships that doesn't have an underscore prefix indicating it's private.
2. `pydub.pyaudioop` is a python implementation of the `audioop` module from the standard library. The motivation for writing it was that, once upon a time, pypy didn't ship the `audioop` module. I contributed our implementation to pypy long ago and they've done significant performance work on it since then. We should probably just drop it now.

So, in short, `pydub.pyaudioop` isn't public. I'd say the logging utils aren't either - I can't think of a good reason why anyone would use pydub's logger.
